gas-service/src/lib.rs

```diff
@@ -0,0 +1,287 @@
+#![no_std]
+
+use crate::events::{
+    AddGasData, AddNativeGasData, GasPaidForContractCallData, NativeGasPaidForContractCallData,
+    RefundedData,
+};
+
+multiversx_sc::imports!();
+
+mod events;
+
+#[multiversx_sc::contract]
+pub trait GasService: events::Events {
+    #[init]
+    fn init(&self, gas_collector: &ManagedAddress) {
+        self.gas_collector().set_if_empty(gas_collector);
+    }
+
+    #[payable("*")]
+    #[endpoint(payGasForContractCall)]
+    fn pay_gas_for_contract_call(
+        &self,
+        destination_chain: ManagedBuffer,
+        destination_address: ManagedBuffer,
+        payload: ManagedBuffer,
+        refund_address: ManagedAddress,
+    ) {
+        let (gas_token, gas_fee_amount) = self.call_value().single_fungible_esdt();
```

Comment:
You could squash the two functions by using the `egld_or_single_esdt` function for Call API.
Also instead of TokenIdentifier, you can use `EgldOrEsdtTokenIdentifier` to manage the merge of the two functions.